sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -0,0 +1,294 @@
+
+### Minor renames
+
+| In v3 | Equivalent in v5 |
+|---|---|
+| `EventPosition.fromStartOfStream()` | `EventPosition.earliest()` |
+| `EventPosition.fromEndOfStream()` | `EventPosition.latest()` |
+
+## Migration samples
+
+* [Sending events](#migrating-from-partitionsender-or-eventhubclient-to-eventhubproducerasyncclient-for-sending-events)
+* [Receiving events](#migrating-code-from-partitionreceiver-to-eventhubconsumerclient-for-receiving-events)
+* [Receiving events with checkpointing](#migrating-code-from-eventprocessorhost-to-eventhubconsumerclient-for-receiving-events)
+
+### Migrating code from `PartitionSender` or `EventHubClient` to `EventHubProducerAsyncClient` for sending events
+In v3, there were multiple options on how to publish events to an Event Hub.
+
+In v5, this has been consolidated into a more efficient `send(EventDataBatch)` method. Batching merges information from
+multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at
+a time.
+
+The code below assumes all events fit into a single batch. For a more complete example, see sample: [Publishing events
+to specific partition][PublishEventsToSpecificPartition].
+
+So in v3:
+```java
+EventHubClient client = EventHubClient.createFromConnectionStringSync("connection-string-for-an-event-hub",
+    Executors.newScheduledThreadPool(4));
+
+List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()), EventData.create("bar".getBytes()));
+
+CompletableFuture<Void> sendFuture = client.createPartitionSender("my-partition-id")
+    .thenCompose(producer -> {
+        EventDataBatch batch = producer.createBatch();
+        for (EventData event : events) {
+            try {
+                // Assuming all events fit into a single batch. This returns false if it does not.
+                // If that is the case, we'll send the full batch then create another one to continue adding events to.
+                batch.tryAdd(event);
+            } catch (PayloadSizeExceededException e) {
+                System.err.println("Event is too large for batch. Exception: " + e);
+            }
+        }
+
+        return producer.send(batch);
+    });
+
+sendFuture.get();
+```
+
+In v5:
+```java
+List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()), EventData.create("bar".getBytes()));
+
+EventHubProducerAsyncClient producer = new EventHubClientBuilder()
+    .connectionString("connection-string-for-an-event-hub")
+    .buildAsyncProducerClient();
+
+CreateBatchOptions options = new CreateBatchOptions()
+    .setPartitionId("my-partition-id");
+
+Mono<Void> sendOperation = producer.createBatch(options).flatMap(batch -> {
+    for (EventData event : data) {
+        // Assuming all events fit into a single batch. This returns false if it does not.
+        // If that is the case, we'll send the full batch then create another one to continue adding events to.
+        batch.tryAdd(event);
```

Comment:
For consistency would this also have a "here's the failure case block" style to it (like the V3 block above?)

```java
if (!batch.tryAdd(event)){
   // Event is too large for batch
}
```